azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java

```diff
@@ -115,6 +135,30 @@ protected void initExecution() {
         trackMojoExecution(MojoStatus.Start);
     }
 
+    protected void initializeAuthConfiguration() throws MojoFailureException {
+        if (auth == null) {
+            return;
+        }
+        if (StringUtils.isNotBlank(auth.getServerId())) {
+            if (this.settings.getServer(auth.getServerId()) != null) {
+                try {
+                    auth = MavenSettingHelper.getAuthConfigurationFromServer(session, settingsDecrypter, auth.getServerId());
+                } catch (MavenDecryptException e) {
+                    throw new MojoFailureException(e.getMessage());
+                }
+            } else {
+                throw new MojoFailureException(String.format("Unable to get server('%s') from settings.xml.", auth.getServerId()));
+            }
+        }
+
+        final List<ConfigurationProblem> problems = auth.validate();
+        if (problems.stream().anyMatch(problem -> problem.getSeverity() == Severity.ERROR)) {
```

Comment:
warning problems?

Reply:
if there are only warning problems, the validation passes, but still should log the message, will cover this in newer PR.